internal/driver/webhtml.go

```diff
@@ -0,0 +1,555 @@
+        setMode(IDLE)
+        return
+      }
+      panMove(t.clientX, t.clientY)
+      e.preventDefault()
+    } else if (mode == TOUCHZOOM) {
+      // Get two touches; new gap; rescale to ratio.
+      var t1 = findTouch(e.touches, touchid)
+      var t2 = findTouch(e.touches, touchid2)
+      if (t1 == null || t2 == null) return
+      var gap = touchGap(t1, t2)
+      rescale(initScale * gap / initGap, centerPoint)
+      e.preventDefault()
+    }
+  }
+
+  var handleTouchEnd = function(e) {
+    if (mode == TOUCHPAN) {
+      var t = findTouch(e.changedTouches, touchid)
+      if (t == null) return
+      panMove(t.clientX, t.clientY)
+      setMode(IDLE)
+      e.preventDefault()
+      if (!moved) clickHandler(t.target)
+    } else if (mode == TOUCHZOOM) {
+      setMode(IDLE)
+      e.preventDefault()
+    }
+  }
+
+  svg.addEventListener("mousedown", handleScanStart)
+  svg.addEventListener("mouseup", handleScanEnd)
+  svg.addEventListener("touchstart", handleTouchStart)
+  svg.addEventListener("touchmove", handleTouchMove)
+  svg.addEventListener("touchend", handleTouchEnd)
+  svg.addEventListener("wheel", handleWheel, true)
+}
+
+function dotviewer(nodes) {
+  // Elements
+  var detailsButton = document.getElementById("details")
+  var detailsText = document.getElementById("detailtext")
+  var listButton = document.getElementById("list")
+  var disasmButton = document.getElementById("disasm")
+  var resetButton = document.getElementById("reset")
+  var focusButton = document.getElementById("focus")
+  var showButton = document.getElementById("show")
+  var ignoreButton = document.getElementById("ignore")
+  var hideButton = document.getElementById("hide")
+  var search = document.getElementById("searchbox")
+  var graph0 = document.getElementById("graph0")
+  var svg = graph0.parentElement
+
+  var currentRe = ""
+  var selected = new Map()
+  var origFill = new Map()
+  var searchAlarm = null
+  var buttonsEnabled = true
+
+  var handleDetails = function() {
+    if (detailtext.style.display == "block") {
+      detailtext.style.display = "none"
+      detailsButton.innerText = "\u25b7 Details"
+    } else {
+      detailtext.style.display = "block"
+      detailsButton.innerText = "\u25bd Details"
+    }
+  }
+
+  var handleReset  = function() { window.location.href = "/" }
+  var handleList   = function() { navigate("/weblist", "f", true) }
+  var handleDisasm = function() { navigate("/disasm", "f", true) }
+  var handleFocus  = function() { navigate("/", "f", false) }
+  var handleShow   = function() { navigate("/", "s", false) }
+  var handleIgnore = function() { navigate("/", "i", false) }
+  var handleHide   = function() { navigate("/", "h", false) }
+
+  var handleSearch = function() {
+    // Delay processing so a flurry of key strokes is handled once.
+    if (searchAlarm != null) {
+      clearTimeout(searchAlarm)
+    }
+    searchAlarm = setTimeout(doSearch, 300)
+  }
+
+  doSearch = function() {
```

Comment:
Added 'use strict'; very helpful.  I put it just inside my two global functions instead of the top-level, since I presume you had some reason in mind when you mentioned global functions,